tfjs-backend-webgpu/src/shader_preprocessor_wgsl.ts

```diff
@@ -535,8 +576,10 @@ export function generateGetOutputCoords(
   const outRank = outShape.length;
   if (x.length === outRank) {
     const dtype = getCoordsDataTypeWgsl(outRank);
-    const snippet = `fn getOutputCoords(globalId : vec3<u32>) -> ${dtype}{
-      return getCoordsFromFlatIndex(u32(globalId.x));
+    const snippet =
+        `fn getOutputCoords(globalId : vec3<u32>, globalIndex : u32) -> ${
```

Comment:
A second thought, do you think is it better to define `getOutputCoords` like below?
```
fn getOutputCoords(globalId : vec3<u32>, localId : vec3<u32>) -> ${dtype}{
  return getCoordsFromFlatIndex(getGlobalIndex(globalId, localId));
}
```
In this case, we don't need to calculate `globalIndex` in all programs main body and pass it here. We only use it when needed like `x.length === outRank` is satified.

And for `getGlobalIndex`, `workGroupSize` is not necessary since we can get it from program.workGroupsize in this file.

Reply:
This maty result in getGlobalIndex is called many times, example:
https://github.com/tensorflow/tfjs/pull/5519/files#diff-df04afa736a2348237666850c1f5f40b6f17e2bb47c421198302fbf7abb72e65R135

Reply:
Yes, I think we can do some futher optimization about this. However, how about we land this first? Thi swill block my next two PRs. 
> That's a good point. Maybe we should provide a new func `getXXXFromFlatIndex(int index)` instead of `getXXXAtOutCoords()` when it's flat dispatch layout to avoid repeating calculations. Then all issues can be resolved.

@kainino0x  @lina128, PTAL, thanks!